### PR TITLE
Add sample for updating a subscription.

### PR DESCRIPTION
**Pull request by noerog**
Using ack_deadline_seconds as the example. Tests coming in next PR.

### CODE REVIEW

File: pubsub/cloud-client/subscriber.py [hunk 1]
```diff
@@ -89,6 +89,35 @@ def delete_subscription(project, subscription_name):
...
+    """
+    Updates an existing Pub/Sub subscription's ackDeadlineSeconds
+    from 10 seconds (default). Note that certain properties of a
+    subscription, such as its topic, are not modifiable.
+    """
+    subscriber = pubsub_v1.SubscriberClient()
+    subscription_path = subscriber.subscription_path(
+        project, subscription_name)
+
+    subscription = {
```
Comment (jonparrott) [inline pubsub/cloud-client/subscriber.py]: in the other samples we actually use the real types over dictionaries, so maybe make this `subscription = pubsub_v1.types.Subscription(name=..., ack_deadline_seconds=...)`?

Comment (noerog) [reply pubsub/cloud-client/subscriber.py]: Done.

**Comment by jonparrott** [comment]
Thanks, @noerog!

File: pubsub/cloud-client/subscriber.py [hunk 2]
```diff
@@ -89,6 +89,34 @@ def delete_subscription(project, subscription_name):
     print('Subscription deleted: {}'.format(subscription_path))
 
 
+def update_subscription(project, subscription_name, ack_deadline_seconds):
+    """
+    Updates an existing Pub/Sub subscription's ackDeadlineSeconds
+    from 10 seconds (default). Note that certain properties of a
+    subscription, such as its topic, are not modifiable.
```
Comment (foresmac) [inline pubsub/cloud-client/subscriber.py]: Are the fields that are mutable and non mutable documented anywhere?

**Pull request merged**